hyperscale/Dockerfile

```
@@ -0,0 +1,73 @@
+FROM postgres:14.0
+ARG VERSION=10.2.2
+LABEL maintainer="Citus Data https://citusdata.com" \
+      org.label-schema.name="Citus" \
+      org.label-schema.description="Scalable PostgreSQL for multi-tenant and real-time workloads" \
+      org.label-schema.url="https://www.citusdata.com" \
+      org.label-schema.vcs-url="https://github.com/citusdata/citus" \
+      org.label-schema.vendor="Citus Data, Inc." \
+      org.label-schema.version=${VERSION} \
+      org.label-schema.schema-version="1.0"
+
+ENV CITUS_VERSION ${VERSION}.citus-1
+ENV POSTGRES_DB citus
+
+# install Citus and needed extension on hyperscale
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+       ca-certificates \
+       curl \
+    && curl -s https://install.citusdata.com/community/deb.sh | bash \
+    && apt-get install -y postgresql-$PG_MAJOR-citus-10.2=$CITUS_VERSION \
+                          postgresql-$PG_MAJOR-hll=2.16.citus-1 \
+                          postgresql-$PG_MAJOR-topn=2.4.0 \
+                          postgresql-14-postgis-3 \
+                          postgresql-plperl-14 \
+                          postgresql-contrib \
+                          postgresql-14-partman \
+                          postgresql-14-tdigest \
+                          postgresql-14-mysql-fdw \
+                          postgresql-14-cron \
+    && apt-get purge -y --auto-remove curl \
+    && rm -rf /var/lib/apt/lists/*
+
+
+
+# set shared_preload_libraries to the list of extensions that needs to be loaded when postgresql starts
+RUN echo "shared_preload_libraries='citus,pg_cron,pg_stat_statements'" >> /usr/share/postgresql/postgresql.conf.sample
+
+# set default database name for pg_cron extension
+RUN echo "cron.database_name = 'citus'" >> /usr/share/postgresql/postgresql.conf.sample
+
+# add scripts to run after initdb
+COPY hyperscale/001-db-user-creation.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/002-create_extension_functions.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/003-create_citus_extension.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/004-postgres_replication_origins.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/005-create_postgis_extension.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/006-other_extensions.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/007-pg_cron_pg_partman_setup.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/008-grant_usage_fdw.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/009-other_grants.sql /docker-entrypoint-initdb.d/
+COPY hyperscale/010-stat_statements_reset.sql /docker-entrypoint-initdb.d/
+
+# add health check script
+COPY pg_healthcheck wait-for-manager.sh /
+
+RUN chmod +x /wait-for-manager.sh
+
+# This block is required to be able to define the password for user citus. With the below script citus password can be
+# set using 'CITUS_PASSWORD' env variable set in docker run phase
+RUN apt-get update && apt-get install -y  sudo && adduser postgres sudo
```

You need to run `rm -rf /var/lib/apt/lists/*` after using `apt-get` cli to be able to shrink image size.

```suggestion
RUN apt-get update 
    && apt-get install -y sudo
    && adduser postgres sudo
    && rm -rf /var/lib/apt/lists/*
```

However, I do not really understand why you want postgres to have sudo privileges. It is not really recommended to do this.

---

@marcocitus wanted me to parametize the password change. I need to change the password in docker run phase to be able to replace password that is the sole solution that I could find.

---

An excerpt from [Docker docs](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user)

> Avoid installing or using sudo as it has unpredictable TTY and signal-forwarding behavior that can cause problems. If you absolutely need functionality similar to sudo, such as initializing the daemon as root but running it as non-root, consider using “gosu”.

Please do not install sudo without making sure we are ok with all the consequences that comes with it.

I am 100% sure we do not give sudo permissions to postgres user on HyperScale, and we should mimic the behaviour here as well.

---

Removed sudo and changed ownership folder for migration folder for  postgres user